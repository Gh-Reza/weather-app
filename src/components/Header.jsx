import { NavLink, useLocation } from 'react-router-dom';
import Icon from '../assets/icon.png';
import BackIcon from '../assets/icon50.png';

const Header = () => {
  const location = useLocation();
  const isDetailsPage = location.pathname.startsWith('/details/');
  return (
    <header className="header text-white d-flex justify-content-between align-items-center">
      <div className="d-flex">
        {isDetailsPage && <NavLink to="/"><img src={BackIcon} alt="Back icon" className="back-icon" /></NavLink> }
        <p className="h2">{new Date().getFullYear()}</p>
      </div>
      <div className="left d-flex align-items-center">
        <h1 className="web-title">WeatherAPP</h1>
      </div>
      <div className="right">
        <img src={Icon} alt="" />
      </div>
    </header>
  );
};

export default Header;
