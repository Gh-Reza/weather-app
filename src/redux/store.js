import { configureStore, applyMiddleware } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
}, applyMiddleware);

export default store;
