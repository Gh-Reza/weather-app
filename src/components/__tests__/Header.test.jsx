import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/store';
import '@testing-library/jest-dom/extend-expect';
import Header from '../Header';

describe('Tests the Header component', () => {
  const component = (
    <BrowserRouter>
      <Provider store={store}>
        <Header />
      </Provider>
    </BrowserRouter>
  );

  test('Match the snapshot', () => {
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  test('Should contain an img tag', () => {
    render(component);
    const img = screen.getByRole('img');
    expect(img).toBeInTheDocument();
  });

  test('Should display the text', () => {
    render(component);
    const text = screen.getByText('WeatherAPP');
    expect(text).toBeInTheDocument();
  });
});
