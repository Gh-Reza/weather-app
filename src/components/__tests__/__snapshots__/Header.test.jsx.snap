// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the Header component Match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header text-white d-flex justify-content-between align-items-center"
      >
        <div
          class="left d-flex align-items-center"
        >
          <h1
            class="fs-1"
          >
            WeatherAPP
          </h1>
        </div>
        <div
          class="right"
        >
          <div
            class="search-container"
          >
            <input
              class="input"
              id=""
              name=""
              placeholder="Search"
              type="text"
            />
            <ul
              class="result-box"
            />
          </div>
          <img
            alt=""
            src="icon.png"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header text-white d-flex justify-content-between align-items-center"
    >
      <div
        class="left d-flex align-items-center"
      >
        <h1
          class="fs-1"
        >
          WeatherAPP
        </h1>
      </div>
      <div
        class="right"
      >
        <div
          class="search-container"
        >
          <input
            class="input"
            id=""
            name=""
            placeholder="Search"
            type="text"
          />
          <ul
            class="result-box"
          />
        </div>
        <img
          alt=""
          src="icon.png"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
